### PR TITLE
refactor: [tidy] modernize-use-emplace

### PR DESCRIPTION
**Pull request by maflcko**
Constructing a temporary unnamed object only to copy or move it into a container seems both verbose in code and a strict performance penalty.

Fix both issues via the `modernize-use-emplace` tidy check.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

**Comment by hebasto** [comment]
Can we do something with code like that: https://github.com/bitcoin/bitcoin/blob/db7b5dfcc502a8a81c51f56fe753990ae8b3a202/src/bitcoin-cli.cpp#L325 ?

**Review by hebasto — APPROVED** [review]
ACK dddd734db8140c9ae63cdbefd1f95c9f00ae5b87, I've followed clang-tidy's warnings locally.

In your opinion, which value of the [`IgnoreImplicitConstructors`](https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-emplace.html#cmdoption-arg-IgnoreImplicitConstructors) options is better/safer for us?

**Comment by Sjors** [comment]
utACK dddd734db8140c9ae63cdbefd1f95c9f00ae5b87

> A push function is not able to call explicit constructors. However, if the code compiles with a push function, it should also compile with an emplace function.

Someone on the internet [says](https://stackoverflow.com/a/36919571/313633) that this ability to call explicit constructors can sometimes lead to bugs. But since everything here was first compiled with `push_back`, that's not an issue. Just something to keep an eye on for _new_ uses of `emplace_back`.

**Review by aureleoules — APPROVED** [review]
ACK dddd734db8140c9ae63cdbefd1f95c9f00ae5b87

**Comment by Sjors** [comment]
Example of the CI error:

```
clang-tidy-17 -p=/ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu -quiet -load=/tidy-build/libbitcoin-tidy.so /ci_container_base/ci/scratch/build/bitcoin-x86_64-pc-linux-gnu/src/script/signingprovider.cpp
script/signingprovider.cpp:371:41: error: unnecessary temporary object created while calling emplace_back [modernize-use-emplace,-warnings-as-errors]
  371 |     if (track) node.leaves.emplace_back(LeafInfo{.script = std::vector<unsigned char>(script.begin(), script.end()), .leaf_version = leaf_version, .merkle_branch = {}});
      |                 
```

File: src/script/signingprovider.cpp [hunk 1]
```diff
@@ -368,6 +368,7 @@ TaprootBuilder& TaprootBuilder::Add(int depth, Span<const unsigned char> script,
     /* Construct NodeInfo object with leaf hash and (if track is true) also leaf information. */
     NodeInfo node;
     node.hash = ComputeTapleafHash(leaf_version, script);
+    // NOLINTNEXTLINE(modernize-use-emplace)
```
Comment (Sjors) [inline src/script/signingprovider.cpp:371]: Can you explain why in the source comment? (at least outside of test and fuzz code)

**Comment by Sjors** [comment]
re-utACK fa05a726c2

File: src/.bear-tidy-config [hunk 2]
```diff
@@ -8,6 +8,8 @@
         "src/crypto/ctaes",
         "src/leveldb",
         "src/minisketch",
+        "src/bench/nanobench.cpp",
+        "src/bench/nanobench.h",
```
Comment (hebasto) [inline src/.bear-tidy-config:12]: nit: They might be ordered lexicographically?

**Review by hebasto — APPROVED** [review]
ACK fa05a726c225dc65dee79367bb67f099ae4f99e6.

**Review by sedited — APPROVED** [review]
ACK fa05a726c225dc65dee79367bb67f099ae4f99e6

**Pull request merged**